cannon/mipsevm/program/patch.go

```diff
@@ -73,16 +72,24 @@ func PatchStack(st mipsevm.FPVMState) error {
 	}
 
 	// init argc, argv, aux on stack
-	storeMem(sp+4*1, 0x42)   // argc = 0 (argument count)
-	storeMem(sp+4*2, 0x35)   // argv[n] = 0 (terminating argv)
-	storeMem(sp+4*3, 0)      // envp[term] = 0 (no env vars)
-	storeMem(sp+4*4, 6)      // auxv[0] = _AT_PAGESZ = 6 (key)
-	storeMem(sp+4*5, 4096)   // auxv[1] = page size of 4 KiB (value) - (== minPhysPageSize)
-	storeMem(sp+4*6, 25)     // auxv[2] = AT_RANDOM
-	storeMem(sp+4*7, sp+4*9) // auxv[3] = address of 16 bytes containing random value
-	storeMem(sp+4*8, 0)      // auxv[term] = 0
+	storeMem(sp+4*0, 1)       // argc = 1 (argument count)
+	storeMem(sp+4*1, sp+4*20) // argv[0]
+	storeMem(sp+4*2, 0)       // argv[1] = terminating
+	storeMem(sp+4*3, sp+4*15) // envp[0] = 0 (no env vars)
+	storeMem(sp+4*4, 0)       // envp[1] = terminating
+	storeMem(sp+4*5, 6)       // auxv[0] = _AT_PAGESZ = 6 (key)
+	storeMem(sp+4*6, 4096)    // auxv[1] = page size of 4 KiB (value) - (== minPhysPageSize)
+	storeMem(sp+4*7, 25)      // auxv[2] = AT_RANDOM
+	storeMem(sp+4*8, sp+4*10) // auxv[3] = address of 16 bytes containing random value
+	storeMem(sp+4*9, 0)       // auxv[term] = 0
 
-	_ = st.GetMemory().SetMemoryRange(sp+4*9, bytes.NewReader([]byte("4;byfairdiceroll"))) // 16 bytes of "randomness"
+	_ = st.GetMemory().SetMemoryRange(sp+4*10, bytes.NewReader([]byte("4;byfairdiceroll"))) // 16 bytes of "randomness"
+
+	envar := append([]byte("memprofilerate=0"), 0x0)
+	_ = st.GetMemory().SetMemoryRange(sp+4*15, bytes.NewReader(envar))
+
+	programName := append([]byte("op-program"), 0x0)
```

Comment:
An assertion, that the env-var that precides this, is indeed 16 bytes + 4 bytes for terminator, would be nice as sanity-check. Or at least a comment about the otherwise seemingly magic offsets.

Reply:
added a comment. The `entry.elf` test already asserts the layout.